libs/traits/src/lib.rs

```diff
@@ -695,3 +695,71 @@ pub mod fees {
 		}
 	}
 }
+
+mod ops {
+	pub use sp_runtime::{
+		traits::{CheckedAdd, CheckedDiv, CheckedMul, CheckedSub},
+		ArithmeticError,
+	};
+
+	/// Performs addition that returns `ArithmeticError::Overflow` instead of
+	/// wrapping around on overflow.
+	pub trait EnsureAdd: CheckedAdd {
+		/// Adds two numbers, checking for overflow. If overflow happens,
+		/// `ArithmeticError::Overflow` is returned.
+		///
+		/// ```
+		/// u32::max().ensure_add(1)?
+		/// ```
+		fn ensure_add(&self, v: &Self) -> Result<Self, ArithmeticError> {
+			self.checked_add(v).ok_or(ArithmeticError::Overflow)
+		}
+	}
+
+	/// Performs subtraction that returns `ArithmeticError::Underflow` instead of
+	/// wrapping around on underflow.
+	pub trait EnsureSub: CheckedSub {
+		/// Subtracts two numbers, checking for overflow. If overflow happens,
+		/// `ArithmeticError::Underflow` is returned.
+		///
+		/// ```
+		/// 0u32.ensure_sub(1)?
+		/// ```
+		fn ensure_sub(&self, v: &Self) -> Result<Self, ArithmeticError> {
+			self.checked_sub(v).ok_or(ArithmeticError::Underflow)
+		}
+	}
+
+	/// Performs multiplication that returns `ArithmeticError::Overflow` instead of
+	/// wrapping around on overflow.
+	pub trait EnsureMul: CheckedMul {
+		/// Multiplies two numbers, checking for overflow. If overflow happens,
+		/// `ArithmeticError::Overflow` is returned.
+		///
+		/// ```
+		/// u32::max().ensure_mul(2)?
+		/// ```
+		fn ensure_mul(&self, v: &Self) -> Result<Self, ArithmeticError> {
+			self.checked_mul(v).ok_or(ArithmeticError::Overflow)
+		}
+	}
+
+	/// Performs division that returns `ArithmeticError::DivisionByZero` instead of
+	/// wrapping around on overflow.
+	pub trait EnsureDiv: CheckedDiv {
+		/// Divides two numbers, checking for overflow. If overflow happens,
+		/// `ArithmeticError::DivisionByZero` is returned.
+		///
+		/// ```
+		/// u32::max().ensure_div(0)?
+		/// ```
+		fn ensure_div(&self, v: &Self) -> Result<Self, ArithmeticError> {
+			self.checked_div(v).ok_or(ArithmeticError::DivisionByZero)
```

Comment:
I leave a tech debt here to fix the case where the division error is due to underflow or overflow. To be honest, I do not understand when it can happen in a division, but the [`checked_div`](https://docs.rs/num-traits/latest/num_traits/ops/checked/trait.CheckedDiv.html#tymethod.checked_div) says that it may happen (maybe with float numbers only?).

Reply:
We can write a small macro and only implement it for unsigned and signed? WDYT?

Who uses floats anyways ^^

Reply:
Where did you get that info from?

Reply:
From the documentation of `checked_div`: *Divides two numbers, checking for underflow, overflow, and division by zero. If any of that happens, None is returned.*

PS. floats are for losers 😎